### DISCUSSION TITLE
Does vitest support checking test file type errors when running tests?

### DISCUSSION THREAD

**Discussion by await-ovo** [vitest-dev/vitest#5338] [Q&A] [answered]
It looks like `pnpm vitest` will passes the test fine when there is a type error in the test file, here is a [example](https://stackblitz.com/edit/vitest-dev-vitest-nztcca?file=test%2Fbasic.test.ts&initialPath=__vitest__/):

![image](https://github.com/vitest-dev/vitest/assets/41503212/dcf160e7-4951-46d5-9da7-b5bdf0d520ca)

I tried `typecheck.enabled: true`, but it doesn't seem to be used to solve this problem,  what I want to know is if there is a way to throw the error when there is a type error in the test file instead of passing the test, like `ts-jest` does.

**Comment by await-ovo** [comment]
It seems to work fine after setting `typecheck.include` to `['**\/*.{test,spec}.?(c|m)[jt]s?(x)']`

**Reply by hi-ogawa** [reply, marked as answer]
That will make all your tests into type-only testing, so probably that's not what you want since it will make this test to pass:
https://stackblitz.com/edit/vitest-dev-vitest-muuvf6?file=test%2Fsuite.test.ts

This is a little confusing, but it looks like you need at least one type test to trigger Vitest type checking, so adding something like `empty.test-d.ts` might be a quick way to do that.
https://stackblitz.com/edit/vitest-dev-vitest-mzzt3u?file=test%2Fempty.test-d.ts
